lib/exe/js.php

```diff
@@ -419,9 +419,9 @@ function js_compress($s){
 
             // Only consider deleting whitespace if the signs before and after
             // are not equal and are not an operator which may not follow itself.
-            if ((!$lch || $s[$i+1] == ' ')
+            if (isset($s[$i+1]) && ((!$lch || $s[$i+1] == ' ')
```

Comment:
This isn't functionally equivalent.  plus, is the more sensible test, $i+1 < $slen ?